packages/runtime/src/helpers/functions.ts

```diff
@@ -47,19 +58,16 @@ export const generateFunctions = async (
     ? relative(functionDir, nextServerModuleAbsoluteLocation)
     : undefined
 
-  for (const { route, config, compiled } of apiRoutes) {
-    // Don't write a lambda if the runtime is edge
-    if (isEdgeConfig(config.runtime)) {
-      continue
-    }
-    const apiHandlerSource = await getApiHandler({
-      page: route,
-      config,
+  for (const apiLambda of apiLambdas) {
+    const { functionName, routes, type, includedFiles } = apiLambda
+
+    const apiHandlerSource = getApiHandler({
+      schedule: type === ApiRouteType.SCHEDULED ? routes[0].config.schedule : undefined,
```

Comment:
Nit: A comment here would be nice to point out that `routes` is an array because for the other types it can be multiple and for scheduled ones it's only ever one item. Otherwise `routes[0]` can seem weird

Reply:
added a comment in https://github.com/netlify/next-runtime/pull/2058/commits/223990ec1694c7c733208dd5070fef46eb78c3fc.